packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -101,22 +89,56 @@ export class PersonalizeMiddleware extends MiddlewareBase {
     };
   }
 
-  protected initializeEngageServer(
-    hostName: string,
-    site: SiteInfo,
-    language: string
-  ): EngageServer {
-    const engageServer = initServer({
-      clientKey: this.config.cdpConfig.clientKey,
-      targetURL: this.config.cdpConfig.endpoint,
-      pointOfSale: this.config.getPointOfSale
-        ? this.config.getPointOfSale(site, language)
-        : PosResolver.resolve(site, language),
-      cookieDomain: hostName,
-      forceServerCookieMode: true,
-    });
+  protected async initPersonalizeServer({
+    hostname,
+    sitecoreEdgeContextId,
+    siteName,
+    request,
+    response,
+  }: {
+    hostname: string;
+    sitecoreEdgeContextId: string;
+    siteName: string;
+    request: NextRequest;
+    response: NextResponse;
+  }): Promise<void> {
+    await init(
+      {
+        sitecoreEdgeContextId,
+        siteName,
+        cookieDomain: hostname,
+        enableServerCookie: true,
+      },
+      request,
+      response
+    );
+  }
+
+  protected async personalize(
+    {
+      params,
+      personalizeInfo,
+      language,
+      timeout,
+    }: {
+      personalizeInfo: PersonalizeInfo;
+      params: ExperienceParams;
+      language: string;
+      timeout?: number;
+    },
+    request: NextRequest
+  ) {
+    const personalizationData = {
+      channel: this.config.cdpConfig.channel || 'WEB',
+      currency: this.config.cdpConfig.currency ?? 'USA',
```

Comment:
Believe this should be 'USD'
```suggestion
      currency: this.config.cdpConfig.currency ?? 'USD',
```